scheduler/src/cook/mesos/api.clj

```diff
@@ -975,6 +1025,25 @@
      :exists? (partial retrieve-jobs conn)
      :handle-ok (fn [ctx] (render-jobs-for-response conn framework-id retrieve-sandbox-directory-from-agent ctx))}))
 
+(defn read-jobs-handler
+  [conn framework-id is-authorized-fn retrieve-sandbox-directory-from-agent handle-ok-fn]
+  (base-cook-handler
+    {:allowed-methods [:get]
+     :malformed? job-request-malformed?
+     :allowed? (partial job-request-allowed? conn is-authorized-fn)
+     :exists? (partial jobs-exist? conn)
+     :handle-ok handle-ok-fn}))
+
+(defn read-jobs-handler-multiple
+  [conn framework-id is-authorized-fn retrieve-sandbox-directory-from-agent]
+  (read-jobs-handler conn framework-id is-authorized-fn retrieve-sandbox-directory-from-agent
+                     (partial render-jobs-for-response conn framework-id retrieve-sandbox-directory-from-agent)))
```

Comment:
Nitpick: `->>` would help with readability

Reply:
I agree that this is a bit verbose, but I'd prefer that you refactor (and probably rename) `read-jobs-handler` to behave more like `base-cook-handler`; i.e., we should "curry" `read-jobs-handler` so that the outer function only takes the `handle-ok-fn` argument.

Reply:
I tried to make it a little more clear. I didn't find a good way to make the outer function only take the `handle-ok-fn` argument. Let me know what you think